apps/web/vibes/soul/sections/product-detail/product-detail-form.tsx

```diff
@@ -154,6 +164,18 @@ function FormField({
     [setParam, controls],
   );
 
+  const handleOnOptionMouseEnter = (value: string) => {
+    if (prefetch) {
+      const optionSearchParams = new URLSearchParams(searchParams.toString());
+
+      optionSearchParams.set(String(field.name), String(value));
+
+      const newUrl = `${pathname}?${optionSearchParams.toString()}`;
+
+      router.prefetch(newUrl);
```

Comment:
Is the idea here that we're doing what will happen when an option is selected on mouse enter? If so, we shouldn't use `URLSearchParams` and should use `nuqs` to keep things consistent (i.e., use the same parsers, setters, etc.)

Reply:
Not sure I follow. We don't want to set params here, we only want to know the url we need to prefetch. Setting params in nuqs would trigger a redirect no?

Reply:
`nuqs` provides serializers to turn values into URL search params that can share the same parsers we're using in `useQueryState`. More detail [here](https://nuqs.47ng.com/docs/utilities#serializer-helper)

Reply:
Updated to use nuqs 👍